### PR TITLE
fix: replace all distinguished types by mixed types.

### PR DESCRIPTION
**Pull request by wanghan-iapcm**

### CODE REVIEW

File: deepmd/dpmodel/atomic_model/linear_atomic_model.py [hunk 1]
```diff
@@ -50,13 +50,11 @@ def __init__(
...
         super().__init__()
         self.models = models
-        self.distinguish_type_list = [
-            model.distinguish_types() for model in self.models
-        ]
+        self.mixed_types_list = [model.mixed_types() for model in self.models]
 
-    def distinguish_types(self) -> bool:
+    def mixed_types(self) -> bool:
         """If distinguish different types by sorting."""
```
Comment (iProzd) [inline deepmd/dpmodel/atomic_model/linear_atomic_model.py:56]: docstr may be 'If ignore differences in types when sorting.'

File: deepmd/dpmodel/fitting/invar_fitting.py [hunk 2]
```diff
@@ -126,7 +126,7 @@ def __init__(
         layer_name: Optional[List[Optional[str]]] = None,
         use_aparam_as_mask: bool = False,
         spin: Any = None,
-        distinguish_types: bool = False,
+        mixed_types: bool = False,
```
Comment (iProzd) [inline deepmd/dpmodel/fitting/invar_fitting.py:129]: Update the default value.

File: deepmd/pt/model/descriptor/dpa1.py [hunk 3]
```diff
@@ -108,11 +108,17 @@ def get_dim_out(self) -> int:
     def get_dim_emb(self) -> int:
         return self.se_atten.dim_emb
 
-    def distinguish_types(self) -> bool:
-        """Returns if the descriptor requires a neighbor list that distinguish different
-        atomic types or not.
+    def mixed_types(self) -> bool:
+        """If true, the discriptor
+        1. assumes total numbe of atoms aligned across frames;
```
Comment (iProzd) [inline deepmd/pt/model/descriptor/dpa1.py:113]: 'numbe' typo

File: deepmd/dpmodel/model/make_model.py [hunk 4]
```diff
@@ -111,7 +111,7 @@ def call(
                 nloc,
                 self.get_rcut(),
                 self.get_sel(),
-                distinguish_types=self.distinguish_types(),
+                distinguish_types=not self.mixed_types(),
```
Comment (iProzd) [inline deepmd/dpmodel/model/make_model.py:114]: key not changed for `build_neighbor_list`

Comment (iProzd) [reply deepmd/dpmodel/model/make_model.py:114]: maybe it's ok here

File: deepmd/pt/model/descriptor/dpa1.py (hunk 3 above)
Comment (njzjz) [reply deepmd/pt/model/descriptor/dpa1.py:113]: Same issue below

File: deepmd/pt/model/descriptor/dpa1.py [hunk 5]
```diff
@@ -108,11 +108,17 @@ def get_dim_out(self) -> int:
...
+        2. requires a neighbor list that does not distinguish different atomic types.
+
+        If false, the discriptor
+        1. assumes total numbe of atoms of each atom type aligned across frames;
```
Comment (njzjz) [inline deepmd/pt/model/descriptor/dpa1.py:117]: I think it also requires the atom type to be aligned, not only the number.

File: deepmd/pt/model/descriptor/dpa1.py [hunk 6]
```diff
@@ -108,11 +108,17 @@ def get_dim_out(self) -> int:
...
+        2. requires a neighbor list that distinguishes different atomic types.
```
Comment (njzjz) [inline deepmd/pt/model/descriptor/dpa1.py:118]: TF requires both central and neighbor atoms to be sorted by atom types. Does PT have this requirement?

File: deepmd/pt/model/descriptor/dpa1.py (hunk 5 above)
Comment (wanghan-iapcm) [reply deepmd/pt/model/descriptor/dpa1.py:117]: what is the different ? i did not get.

File: deepmd/pt/model/descriptor/dpa1.py (hunk 6 above)
Comment (wanghan-iapcm) [reply deepmd/pt/model/descriptor/dpa1.py:118]: pt does not support "type two side", i.e. it only use different nets for different neighbor types, but not for different center atom types, thus the center atoms are not assumed to be sorted.

File: deepmd/pt/model/descriptor/dpa1.py (hunk 5 above)
Comment (njzjz) [reply deepmd/pt/model/descriptor/dpa1.py]: Is this allowed?

Frame 0: 0 0 0 1 1 1
Frame 1: 1 1 1 0 0 0

**Review by njzjz — APPROVED** [review]

**Review by anyangml — APPROVED** [review]

File: deepmd/pt/model/descriptor/dpa1.py (hunk 5 above)
Comment (wanghan-iapcm) [reply deepmd/pt/model/descriptor/dpa1.py]: yes see https://github.com/deepmodeling/deepmd-kit/pull/3295

**Pull request merged**